### ISSUE TITLE
Headless client won't startup after latest update

### ISSUE THREAD

**Issue by ScarsTRF** [Yellow-Dog-Man/Resonite-Issues#1134]
### Describe the bug?

The headless client can't find something for opus assembly

### To Reproduce

- Download latest headless client update
- Load the headless client
- Instantly the client faults

### Expected behavior

Headless client should launch without issue

### Resonite Version Number

Beta 2024.1.12.1336

### What Platforms does this occur on?

Linux

### Log Files

[73e7bde3dd4a - 2024.1.12.1336 - 2024-01-12 17_42_56.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/13925065/73e7bde3dd4a.-.2024.1.12.1336.-.2024-01-12.17_42_56.log)

### Reporters

ScarsTRF | Discord: scarstrf

**Comment by BigRedWolfy** [comment]
Can confirm mentor gathering workshop headless ain't loading for the same reasons

[93ca53d4-6ad1-4990-8f37-43beae8bf237 - 2024.1.12.1336 - 2024-01-12 23_59_07.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/13925152/93ca53d4-6ad1-4990-8f37-43beae8bf237.-.2024.1.12.1336.-.2024-01-12.23_59_07.log)
[93ca53d4-6ad1-4990-8f37-43beae8bf237 - 2024.1.12.1336 - 2024-01-12 23_59_23.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/13925153/93ca53d4-6ad1-4990-8f37-43beae8bf237.-.2024.1.12.1336.-.2024-01-12.23_59_23.log)

The environment I'm running for my headless is about the same as ScarsTRF

**Comment by Frooxius** [comment]
Fixed in 2024.1.13.8, thanks!

**Comment by sveken** [comment]
The latest update now has a different error message and is now broken for me. 
[0c033a41fe44 - 2024.1.13.8 - 2024-01-13 11_22_46.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/13925202/0c033a41fe44.-.2024.1.13.8.-.2024-01-13.11_22_46.log)

**Comment by sveken** [comment]
Reuploading log file,

But the latest hotfix is still broken for me and has a n
[0c033a41fe44 - 2024.1.13.8 - 2024-01-13 11_26_38.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/13925209/0c033a41fe44.-.2024.1.13.8.-.2024-01-13.11_26_38.log)
ew error

**Comment by ProbablePrime** [comment]
Looks like the same issue as opus but for assimp.

**Comment by Frooxius** [comment]
It doesn't seem like that. It continues past that error, assimp is not blocking to startup.

**Comment by stiefeljackal** [comment]
Brotli seems to be the culprit since it is present in all logs.

**Comment by BigRedWolfy** [comment]
I have a headless run only once. It was updated to version Beta 2024.1.12.1336 and after the hotfix it did run, but on second startup I was getting the same issue.

Collecting the logs atm

**Comment by BigRedWolfy** [comment]
[93ca53d4-6ad1-4990-8f37-43beae8bf237 - 2024.1.13.8 - 2024-01-13 00_22_00.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/13925225/93ca53d4-6ad1-4990-8f37-43beae8bf237.-.2024.1.13.8.-.2024-01-13.00_22_00.log)
[93ca53d4-6ad1-4990-8f37-43beae8bf237 - 2024.1.13.8 - 2024-01-13 00_22_16.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/13925226/93ca53d4-6ad1-4990-8f37-43beae8bf237.-.2024.1.13.8.-.2024-01-13.00_22_16.log)
[93ca53d4-6ad1-4990-8f37-43beae8bf237 - 2024.1.13.8 - 2024-01-13 00_32_40.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/13925227/93ca53d4-6ad1-4990-8f37-43beae8bf237.-.2024.1.13.8.-.2024-01-13.00_32_40.log)
[93ca53d4-6ad1-4990-8f37-43beae8bf237 - 2024.1.13.8 - 2024-01-13 00_32_59.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/13925228/93ca53d4-6ad1-4990-8f37-43beae8bf237.-.2024.1.13.8.-.2024-01-13.00_32_59.log)
[93ca53d4-6ad1-4990-8f37-43beae8bf237 - 2024.1.13.8 - 2024-01-13 00_19_52.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/13925229/93ca53d4-6ad1-4990-8f37-43beae8bf237.-.2024.1.13.8.-.2024-01-13.00_19_52.log)
[93ca53d4-6ad1-4990-8f37-43beae8bf237 - 2024.1.13.8 - 2024-01-13 00_20_58.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/13925230/93ca53d4-6ad1-4990-8f37-43beae8bf237.-.2024.1.13.8.-.2024-01-13.00_20_58.log)
[93ca53d4-6ad1-4990-8f37-43beae8bf237 - 2024.1.13.8 - 2024-01-13 00_21_13.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/13925231/93ca53d4-6ad1-4990-8f37-43beae8bf237.-.2024.1.13.8.-.2024-01-13.00_21_13.log)

**Comment by stiefeljackal** [comment]
> It doesn't seem like that. It continues past that error, assimp is not blocking the startup.

You are correct. The assimp error is a thing for the Linux Headless build, so it does not block the startup. I'm thinking it is Brotli that could be the cause since that is used to load the settings for the account at the start.

**Comment by BigRedWolfy** [comment]
93ca53d4-6ad1-4990-8f37-43beae8bf237 - 2024.1.13.8 - 2024-01-13 00_19_52.log is the first startup after hotfix I think

**Comment by Frooxius** [comment]
I'm checking. It does seem the Brotli is having issues, but this one is a bit less clear, because the library does seem to be present there.

**Comment by stiefeljackal** [comment]
Another thing common in those logs. Around Line 320, it looks like Brotli is failing to sync R-Settings because R-Settings is larger than 1 MB. The R-Settings size somehow became 1.31 MB

**Comment by BigRedWolfy** [comment]
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/1814393/7e355320-5095-4baf-b043-3a899736f4f9)
Yep

**Comment by bredo228** [comment]
> I'm checking. It does seem the Brotli is having issues, but this one is a bit less clear, because the library does seem to be present there.
> 
> Can you check if `brolib_x64.so` is present in your Linux headless install?

```
root@glaceon:/var/lib/pterodactyl/volumes/93ca53d4-6ad1-4990-8f37-43beae8bf237# find . -iname brolib* -exec md5sum {} \;
9d4da89ec6cb8909934459b9d3de0b45  ./Resonite_Data/Plugins/brolib_x64.so
9d4da89ec6cb8909934459b9d3de0b45  ./Headless/brolib_x64.so
```

Same headless @BigRedWolfy is looking at.

**Comment by bredo228** [comment]
Threw a brotli related exception into the console on shutdown.

Resonite log and copy of the console log:

[resonite-test-debian - 2024.1.13.8 - 2024-01-13 13_54_38.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/13925304/resonite-test-debian.-.2024.1.13.8.-.2024-01-13.13_54_38.log)

[new 16.txt](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/13925303/new.16.txt)

**Comment by Frooxius** [comment]
I have a fix coming: https://github.com/Yellow-Dog-Man/brotli.net/pull/2

Issue is that the the CI/CD update switched to .NET 4.6 version instead of .NET Standard 2.0 of the library. This version for some reason was hardcoded to always assume it's in Windows, instead of detecting this dynamically.

Once the library is recompiled, I'll push an update.

**Comment by stiefeljackal** [comment]
Yep. I was just going to mention that due to the following:
`System.NotSupportedException: Unable to find library brolib_x64.dll`

Those that were affected with the larger R-Settings size will have issues with syncing due to the 1 MB limit.

**Comment by Frooxius** [comment]
@stiefeljackal There's no actual 1 MB limit, that's a bug in Brotli. It'll get resolved with this.

**Comment by stiefeljackal** [comment]
Cool. That might help resolve #941 since that user's R-Settings size is over 1 MB as well.

**Comment by Frooxius** [comment]
@stiefeljackal I'd recommend posting in that issue if there's another to keep this one focused on the startup issue.

**Comment by Frooxius** [comment]
Should be fixed with 2024.1.13.83, thanks!

Reopen if there's still an issue with startup. For other issues, please make a new issue or post into appropriate one.

**Comment by bredo228** [comment]
appears to be working now!

enjoy your food @Frooxius
